types/ApiParams.ts

```diff
@@ -0,0 +1,17 @@
+import { IsEmail, IsUUID, IsHexadecimal, Length } from 'class-validator';
+import { Uuid } from '.';
+
+export class ValidEmail{
```

Comment:
Since these classes are for validating the API requests, they belong in `/api/validators/GenericRequests.ts`, not in the `types` folder. `types` is for defining API interfaces, like specifications that an API implementation should conform to, but these validators are part of the API implementation not its specification. Can explain further it that doesn't make sense.